configs/recommended.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const globals = require('globals');
+const react = require('../index');
+const legacy = require('../legacy');
+
+module.exports = {
+  files: ['**/*.{js,jsx,ts,tsx}'],
+  globals: Object.assign({}, globals.serviceworker, globals.browser),
```

Comment:
We don’t specify this now, why would we add it? Environment-specific things should be only in the top level. For example, most people don’t use service workers (and shouldn’t), so it would be harmful to include here; and some react codebases are server-only.

Reply:
> things only in the top level.

Reply:
> things only in the top level.

The new system is all about a concept like `overrides` by **chaining**.
So there is no such a 'top' level, actually.
Compared to the legacy, configs aggressively specify their preference, and override each other, then finally the user has to override by their preference at the end of the chain.

For example, this snippet from the official blog post shows the design intention.

<img width="891" alt="스크린샷 2022-09-12 오후 12 14 11" src="https://user-images.githubusercontent.com/28584151/189567309-7234c8b0-5a93-44fb-b68a-71d9576e086b.png">

The second config's `semi` rule is overridden by the third. Not only rules, but everything can be overridden.

A sharable`customConfig` is just dropped as-is at the first place of the array.
If `eslint-plugin-react/all` or `eslint-plugin-react/recommended` are inserted as-is like that way, and if the object does not have `files` property preconfigured, the `files` would be `['**/*.js']` by default in the new system, which users might think unintuitive.

So the user would have to almost always destruct the config.

```js
import react from `eslint-plugin-react/recommended`

export default [
  {
    files: ['**/*.{js,jsx,ts,tsx}'],
    ...react
  },
  {
  // ... 
  }
]
```

Instead of,


```js
import react from `eslint-plugin-react/recommended`

export default [
  react,
  {
  // ... 
  }
]
```

This is not a big deal, but the point is it might become a boilerplate in many cases.

How do you think?

IMHO,
- `globals`: I respect your opinion. I'd remove it.
- `files`: I think it's good to be the default. If you still think differently, I'd remove it.

Reply:
The top level means the top-level application - the end user.

Reply:
So this comment was about `globals`, but personally I’d remove files too. I think this is a bad design choice, and end users should be the only ones to ever put in filepaths, because they’re the only ones that can know how their codebase is organized.

Reply:
Also, why would that files glob be unintuitive for the user? It’s the default behavior of eslint already, for years.

Reply:
> Also, why would that files glob be unintuitive for the user? It’s the default behavior of eslint already, for years.
Because by default, if we don't preconfigure `files`, it would be `['**/*.js']`, while majority of users use react with .jsx or .tsx. That's why I thought it'd be unintuitive default for users. 
Do you still want to remove `files`?

Reply:
I see what you're saying, but yes, I think this should not include `files`.

Reply:
Okay.